shared-components/AlertBuilder.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useRef, useState } from "react";
+import ReactDOM from "react-dom";
+import * as Stitches from "@stitches/react";
+
+// import { styled } from "@/shared-components/stitches.config";
+import { styled, css, Layout } from "@/shared-components/index";
+
+import Button from "./Button";
+import Span from "./Span";
+
+const StyledAlert = styled("div", {
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  width: "100vw",
+  height: "100vh",
+  position: "fixed",
+  left: 0,
+  top: 0,
+  zIndex: 999,
+  backgroundColor: "rgba(0,0,0,0.6)",
+  overflow: "hidden",
+});
+
+const AlertOverlay = styled(Layout.ColCenter, {
+  backgroundColor: "$white",
+  borderRadius: "8px",
+  margin: "0 16px",
+});
+
+interface AlertOptions {
+  message: React.ReactNode;
+  onClose?: () => void;
+  buttonMessage: React.ReactNode;
+  css?: Stitches.CSS;
+  buttonCSS?: Stitches.CSS;
+}
+
+class AlertBuilder {
+  private rootElement?: HTMLDivElement;
+  private lastTargetElement?: HTMLDivElement;
+
+  public show = (options: AlertOptions) => {
+    const { message, buttonMessage, onClose, css, buttonCSS } = options;
+    this.renderRootElement();
+    this.lastTargetElement = document.createElement("div");
+    ReactDOM.render(
+      <AlertMessage
+        message={message}
+        onRemove={this.removeRootMessage}
+        buttonMessage={buttonMessage}
+        css={css}
+        onClose={onClose}
+        buttonCSS={buttonCSS}
+      />,
```

Comment:
```suggestion
    this.renderRootElement();
    this.lastTargetElement = document.createElement("div");
    ReactDOM.render(
      <AlertMessage {...options} onRemove={this.removeRootMessage} />,
      this.lastTargetElement
    );
```

이런 방법도 있다는거 알아두면 좋습니당